src/libraries/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.PowMod.cs

```diff
@@ -262,7 +266,11 @@ private static uint PowCore(ulong value, uint power, uint modulus, ulong result)
             Span<uint> valueCopy = (size <= StackAllocThreshold ?
                                    stackalloc uint[StackAllocThreshold]
                                    : valueCopyFromPool = ArrayPool<uint>.Shared.Rent(size)).Slice(0, size);
-            valueCopy.Clear();
+
+            // smallish optimization here:
+            // subsequent operations will copy the elements to the beginning of the buffer,
+            // no need to clear everything
+            valueCopy.Slice(value.Length).Clear();
```

Comment:
Just to confirm, is this clearing everything *after* value.Length?

Reply:
Correct. We don't care about beginning of the buffer because it will be replaced by meaningful content. Therefore, there is no reason to clean the entire buffer (which costs extra CPU cycles).